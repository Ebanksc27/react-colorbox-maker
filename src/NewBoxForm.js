import React, { useState } from 'react';

function NewBoxForm({ addBox }) {
  // Initial state for form inputs
  const [formData, setFormData] = useState({
    width: '',
    height: '',
    backgroundColor: ''
  });

  // Function to generate a unique ID
  const uniqueId = () => {
    return Math.floor(Math.random() * 10000);
  };

  // Handle form submission
  const handleSubmit = e => {
    e.preventDefault();
    addBox({ ...formData, id: uniqueId() });
    setFormData({ width: '', height: '', backgroundColor: '' });
  };

  // Handle changes in form inputs
  const handleChange = e => {
    const { name, value } = e.target;
    setFormData(fData => ({
      ...fData,
      [name]: value
    }));
  };

  return (
    <form onSubmit={handleSubmit}>
      <input 
        type="text"
        name="width"
        value={formData.width}
        onChange={handleChange}
        placeholder="Width"
      />
      <input 
        type="text"
        name="height"
        value={formData.height}
        onChange={handleChange}
        placeholder="Height"
      />
      <input 
        type="text"
        name="backgroundColor"
        value={formData.backgroundColor}
        onChange={handleChange}
        placeholder="Background Color"
      />
      <button type="submit">Add a new box</button>
    </form>
  );
}

export default NewBoxForm;
