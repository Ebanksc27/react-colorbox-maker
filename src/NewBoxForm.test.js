import React from 'react';
import { render } from '@testing-library/react';
import NewBoxForm from './NewBoxForm';

test('renders NewBoxForm component without crashing', () => {
  render(<NewBoxForm />);
});

test('NewBoxForm component snapshot', () => {
  const { asFragment } = render(<NewBoxForm />);
  expect(asFragment()).toMatchSnapshot();
});
