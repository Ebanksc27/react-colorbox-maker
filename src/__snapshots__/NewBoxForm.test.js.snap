// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`NewBoxForm component snapshot 1`] = `
<DocumentFragment>
  <form>
    <input
      name="width"
      placeholder="Width"
      type="text"
      value=""
    />
    <input
      name="height"
      placeholder="Height"
      type="text"
      value=""
    />
    <input
      name="backgroundColor"
      placeholder="Background Color"
      type="text"
      value=""
    />
    <button
      type="submit"
    >
      Add a new box
    </button>
  </form>
</DocumentFragment>
`;
